### ISSUE TITLE
Service Check stanza should support HTTP body

### ISSUE THREAD

**Issue by manveru** [hashicorp/nomad#10084]
### Nomad version
v1.0.4

### Operating system and Environment details
NixOS

### Issue
The `job -> group -> service -> check` stanza doesn't support setting a body for `http` checks.
This should be supported, since Consul has the corresponding option: https://www.consul.io/api/agent/check#body

### Reproduction steps

    ❯ nomad job run -output ./hi.hcl         
    Error getting job struct: Error parsing job file from ./hi.hcl:
    hi.hcl:6,9-13: Unsupported argument; An argument named "body" is not expected here.

### Job file (if appropriate)

```hcl
job "hi" {
  group "hi" {
    service {
      check {
        type = "http"
        body = "hello"
      }
    }
  }
}
```

**Comment by shoenig** [comment]
Thanks for the issue, @manveru! This seems reasonable. It shouldn't be difficult to add as it's mostly plumbing from jobspec to the Consul service registration API, and there's plenty of similar fields to look at.

**Comment by nick96** [comment]
Hey! I've had a go at this in https://github.com/hashicorp/nomad/pull/10186. I'm a bit stuck on updating jobs. I'm guessing it's something in `nomad/structs/diff.go` but I can't figure out exactly what.

**Comment by tgross** [comment]
Closed by #10186 and will ship in Nomad 1.1.0.
